### PR TITLE
Add support for transaction clone symbols

### PR DESCRIPTION
**Pull request by miguelsaldivar**
This should add support for transaction clone symbols, I redid the PR because I was having issues rebasing the new commits against the old PR. If you need me to add anything else let me know!

### CODE REVIEW

File: src/ast.rs [hunk 1]
```diff
@@ -7450,6 +7458,26 @@ impl Parse for SpecialName {
 
                 Ok((SpecialName::JavaResource(resource_names), tail))
             }
+            b"GT" => {
+                match tail.peek() {
```
Comment (khuey) [inline src/ast.rs:7462]: Can you do `tail.next_or(error::Error::UnexpectedEnd)?`

File: src/ast.rs [hunk 2]
```diff
@@ -7450,6 +7458,26 @@ impl Parse for SpecialName {
...
+                    None => return Err(error::Error::UnexpectedEnd),
+                    Some(b'n') => {
+                        let tail = consume(b"n", tail)?;
```
Comment (khuey) [inline src/ast.rs:7465]: and then (b'n', tail) => { NonTransactionClone branch }

File: src/ast.rs [hunk 3]
```diff
@@ -7450,6 +7458,26 @@ impl Parse for SpecialName {
...
+                        let (base, tail) =  Encoding::parse(ctx, subs, tail)?;
+                        Ok((SpecialName::NonTransactionClone(Box::new(base)),
+                            tail,
+                        ))
+                    }
+                    Some(b't') | _ => {
+                        let (_, tail) = tail.split_at(1);
```
Comment (khuey) [inline src/ast.rs:7472]: And (b't', tail) | (_, tail) => { TransactionClone branch}

Also a comment about how libiberty accepts all GT? where ? != n as equivalent to GTt would be good.

File: tests/tests.rs [hunk 4]
```diff
@@ -568,6 +568,14 @@ demangles!(
...
     "decltype ({parm#1}(SkRecords::NoOp())) SkRecord::Record::visit<SkRecords::Draw&>(SkRecords::Draw&) const"
 );
+demangles!(
+    _ZGTtNKSt11logic_error4whatEv,
+    "transaction clone for std::logic_error::what() const"
+);
+demangles!(
+    _ZGTnNKSt11logic_error4whatEv,
+    "non-transaction clone for std::logic_error::what() const"
+);
```
Comment (khuey) [inline tests/tests.rs:578]: Please add a test for a GT? variant.

**Review by khuey — CHANGES_REQUESTED** [review]
Do we need to do `Dx` too? The libiberty test suite has `_Z1fPDxFvvES0_` in it.

File: src/ast.rs (hunk 1 above)
Comment (miguelsaldivar) [reply src/ast.rs]: Done, that looks a lot better. Thanks for the suggestion.

**Comment by miguelsaldivar** [comment]
> Do we need to do `Dx` too? The libiberty test suite has `_Z1fPDxFvvES0_` in it.

So `Dx` is in the code [here](https://github.com/gimli-rs/cpp_demangle/blob/9ead193a66c93bfb30b84c47bdbae2e77cd186ca/src/ast.rs#L4165), we just aren't adding `transaction_safe` in the demangling process. I can probably do a different PR for this.

**Review by khuey — APPROVED** [review]

**Comment by khuey** [comment]
Thanks!

**Pull request merged**